compiler/noirc_frontend/src/ast/statement.rs

```diff
@@ -97,6 +101,9 @@ impl StatementKind {
             // A semicolon on a for loop is optional and does nothing
             StatementKind::For(_) => self,
 
+            // No semicolon needed for a resolved statement
+            StatementKind::Interned(_) => self,
```

Comment:
Is this true? What if we have an interned `StatementKind::Expression`?

Reply:
I guess we can wrap it in `Semi` but never produce an error, right?

Reply:
Ah, only if it's not the last statement in the block.

Reply:
Sounds good to me, agreed we should avoid errors here.

Reply:
Ah, the problem is that `Semi` wraps an expression, and here we only have a statement... I don't know how to solve this.

What's the issue if there are two statements in a row inside a block but the first one isn't semi? I just tried a program like that and it seems to work fine.

I'm also not sure how to handle this if we wanted to because at this point we don't know what the interned expression expands to.

Reply:
The only issue with an expression in a block lacking a semicolon is a potential type warning that it expected a Unit type. E.g. `3; 4` should issue this warning IIRC.

I'm more concerned about the case where `4;` is the last expression in a block and we can't add the semicolon because `4` is interned. If we can't fix it though then we can't fix it. We can address it later, I don't think it should block this PR.